### PR TITLE
Migrate controller tests to use BDD testing style

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
Migrate to use the kubebuilder BDD testing style

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. This is the refactoring to prepare for v1beta1 PR.

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/testing/envtest_setup.go [hunk 1]
```diff
@@ -43,6 +44,7 @@ func SetupEnvTest() *envtest.Environment {
 			filepath.Join("..", "..", "..", "config", "crd"),
 			filepath.Join("..", "..", "..", "test", "crds"),
 		},
+		UseExistingCluster: &useExistingCluster,
 	}
```
Comment (ellistarn) [inline pkg/testing/envtest_setup.go:48]: You can use proto.bool for quick primitive booleans

File: pkg/controller/inferenceservice/controller.go [hunk 2]
```diff
@@ -51,100 +50,38 @@ import (
...
 
-var _ reconcile.Reconciler = &ReconcileService{}
+var _ reconcile.Reconciler = &InferenceServiceReconciler{}
 
 // ReconcileService reconciles a Service object
-type ReconcileService struct {
+type InferenceServiceReconciler struct {
 	client.Client
-	scheme   *runtime.Scheme
+	Log      logr.Logger
```
Comment (ellistarn) [inline pkg/controller/inferenceservice/controller.go:69]: What's the goal with this new logging pattern

Comment (yuzisun) [reply pkg/controller/inferenceservice/controller.go:69]: to void the global logger variable which can cause conflicts within package

File: pkg/testing/envtest_setup.go (hunk 1 above)
Comment (yuzisun) [reply pkg/testing/envtest_setup.go:48]: oh nice! I have been looking for a quick way to do for long time!

**Pull request merged**